File: university/src/main/java/com/gutorov/university/entity/ProgramEntity.java
package com.gutorov.university.entity;

public class ProgramEntity extends BaseEntity {
    private String name;

    public ProgramEntity() {
        super();
    }

    public ProgramEntity(String name) {
        this();
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    @Override
    public String toString() {
        return "ProgramEntity{" +
                "id=" + id +
                ", name='" + name + '\'' +
                '}';
    }
}

File: src/main/java/com/gutorov/university/entity/ProgramEntity.kt
package com.gutorov.university.entity

import jakarta.persistence.*

@Entity
@Table(name = "programs")
class ProgramEntity(
    @Column(nullable = false, length = 512) var name: String
) : BaseEntity() {
    @OneToMany(mappedBy = "_program", cascade = [CascadeType.ALL], orphanRemoval = true)
    var applications: MutableSet<ApplicationEntity> = hashSetOf()
        private set

    fun addApplication(application: ApplicationEntity) = applications.add(application.also { it.program = this })
    fun removeApplication(application: ApplicationEntity) = applications.remove(application.also { it.program = null })
}
